panic: interface conversion during syn

got 
```
panic: interface conversion: interface {} is *uint64, not uint64

goroutine 259493 [running]:
github.com/prysmaticlabs/prysm/beacon-chain/sync.(*Service).goodbyeRPCHandler(0xc000015800, 0x1aa52a0, 0xc01aeaa0c0, 0x1425f60, 0xc05d4260f8, 0x1abc5e0, 0xc052c9f320, 0x0, 0x0)
        beacon-chain/sync/rpc_goodbye.go:30 +0x544
github.com/prysmaticlabs/prysm/beacon-chain/sync.(*Service).registerRPC.func1(0x1abc5e0, 0xc052c9f320)
        beacon-chain/sync/rpc.go:88 +0xb6e
github.com/libp2p/go-libp2p/p2p/host/basic.(*BasicHost).SetStreamHandler.func1(0xc052a8f5c0, 0x24, 0x7f44659f7a48, 0xc052c9f320, 0x24, 0xc0014a5140)
        external/com_github_libp2p_go_libp2p/p2p/host/basic/basic_host.go:397 +0x9d
created by github.com/libp2p/go-libp2p/p2p/host/basic.(*BasicHost).newStreamHandler
        external/com_github_libp2p_go_libp2p/p2p/host/basic/basic_host.go:295 +0x6a2
```
 during sync with --initial-sync-cache-state.

--- a/beacon-chain/sync/rpc_goodbye.go
+++ b/beacon-chain/sync/rpc_goodbye.go
@@ -2,7 +2,6 @@ package sync
 
 import (
 	"context"
-	"errors"
 	"fmt"
 	"time"
 

--- a/beacon-chain/sync/rpc_goodbye.go
+++ b/beacon-chain/sync/rpc_goodbye.go
@@ -2,6 +2,7 @@ package sync
 
 import (
 	"context"
+	"errors"
 	"fmt"
 	"time"
 
@@ -27,7 +28,10 @@ func (r *Service) goodbyeRPCHandler(ctx context.Context, msg interface{}, stream
 	defer cancel()
 	setRPCStreamDeadlines(stream)
 
-	m := msg.(uint64)
+	m, ok:= msg.(uint64)
+	if !ok {
+		return fmt.Errorf("wrong message type for goodbye, got %T, wanted uint64", msg)
+	}
 	log := log.WithField("Reason", goodbyeMessage(m))
 	log.WithField("peer", stream.Conn().RemotePeer()).Info("Peer has sent a goodbye message")
 	// closes all streams with the peer

